pkg/agent/plugin/svidstore/aws/aws.go

```diff
@@ -0,0 +1,289 @@
+	}
+
+	if config.SecretAccessKey == "" {
+		config.SecretAccessKey = p.hooks.getenv("AWS_SECRET_ACCESS_KEY")
+	}
+
+	if config.Region == "" {
+		return nil, status.Error(codes.InvalidArgument, "region is required")
+	}
+
+	smClient, err := p.hooks.newClient(ctx, config.SecretAccessKey, config.AccessKeyID, config.Region)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to create secrets manager client: %v", err)
+	}
+
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	p.smClient = smClient
+
+	return &configv1.ConfigureResponse{}, nil
+}
+
+// PutX509SVID puts the specified X509-SVID in the configured AWS Secrets Manager
+func (p *SecretsManagerPlugin) PutX509SVID(ctx context.Context, req *svidstorev1.PutX509SVIDRequest) (*svidstorev1.PutX509SVIDResponse, error) {
+	opt, err := optionsFromSecretData(req.Metadata)
+	if err != nil {
+		return nil, err
+	}
+
+	// Use arn if it is configured if not use name
+	secretID := opt.name
+	if opt.arn != "" {
+		secretID = opt.arn
+	}
+
+	// Encode the secret from a 'workload.X509SVIDResponse'
+	secret, err := svidstore.SecretFromProto(req)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "failed to parse request: %v", err)
+	}
+
+	secretBinary, err := json.Marshal(secret)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to parse payload: %v", err)
+	}
+
+	// Call DescribeSecret to retrieve the details of the secret
+	// and be able to determine if the secret exists
+	secretDesc, err := p.smClient.DescribeSecret(ctx, &secretsmanager.DescribeSecretInput{
+		SecretId: aws.String(secretID),
+	})
+	if err != nil {
+		var resourceNorFoundErr *types.ResourceNotFoundException
+		if errors.As(err, &resourceNorFoundErr) {
+			// Secret not found, creating one with provided `name`
+			resp, err := createSecret(ctx, p.smClient, secretBinary, opt)
+			if err != nil {
+				return nil, err
+			}
+			p.log.With("version_id", aws.StringValue(resp.VersionId)).With("arn", aws.StringValue(resp.ARN)).With("name", aws.StringValue(resp.Name)).Debug("Secret created")
+
+			return &svidstorev1.PutX509SVIDResponse{}, nil
+		}
+
+		// It must not happens.
+		return nil, status.Errorf(codes.Internal, "failed to describe secret: %v", err)
+	}
+
+	// Validate if secret is deleted, and restore it
+	if secretDesc.DeletedDate != nil {
+		resp, err := p.smClient.RestoreSecret(ctx, &secretsmanager.RestoreSecretInput{
+			SecretId: aws.String(secretID),
+		})
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to restore secret %q: %v", secretID, err)
+		}
+		p.log.With("arn", aws.StringValue(resp.ARN)).With("name", aws.StringValue(resp.Name)).Debug("Secret restored")
+	}
+
+	// Validate that the secret has the 'spire-svid' tag. This tag is used to distinguish the secrets
+	// that have SVID information handled by SPIRE
+	if ok := validateTag(secretDesc.Tags); !ok {
+		return nil, status.Error(codes.InvalidArgument, "secret does not contain the 'spire-svid' tag")
+	}
```

Comment:
I think that this validation should probably be done by a function that can be called here and also in DeleteX509SVID.